neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/embedded/impl/EmbeddedNeo4jGraphDatabaseFactory.java

```diff
@@ -7,23 +7,38 @@
 package org.hibernate.ogm.datastore.neo4j.embedded.impl;
 
 import java.io.File;
+import java.lang.invoke.MethodHandles;
+import java.lang.reflect.Field;
 import java.net.URL;
 import java.util.HashMap;
 import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicBoolean;
 
+import org.hibernate.HibernateException;
 import org.hibernate.ogm.datastore.neo4j.Neo4jProperties;
+import org.hibernate.ogm.datastore.neo4j.logging.impl.Log;
+import org.hibernate.ogm.datastore.neo4j.logging.impl.LoggerFactory;
 import org.hibernate.ogm.datastore.neo4j.spi.GraphDatabaseServiceFactory;
 import org.hibernate.ogm.util.configurationreader.spi.ConfigurationPropertyReader;
+
 import org.neo4j.graphdb.GraphDatabaseService;
 import org.neo4j.graphdb.factory.GraphDatabaseBuilder;
 import org.neo4j.graphdb.factory.GraphDatabaseFactory;
+import org.neo4j.kernel.api.exceptions.TransactionFailureException;
+import org.neo4j.kernel.impl.factory.GraphDatabaseFacade;
+import org.neo4j.kernel.internal.GraphDatabaseAPI;
 
 /**
  * Contains methods to create a {@link GraphDatabaseService} for the embedded Neo4j.
  *
  * @author Davide D'Alto &lt;davide@hibernate.org&gt;
  */
 public class EmbeddedNeo4jGraphDatabaseFactory implements GraphDatabaseServiceFactory {
+	public static final int WAITING_TIME_MS = 100;
+
+	private static final Map<String, FactoryHolder> GRAPH_DATABASE_SERVICE_MAP = new ConcurrentHashMap<>();
```

Comment:
It is really necessary to synchronize use:
1. ConcurrentHashMap
2. AtomicBoolean
3. synchronized 
together ?

Reply:
HI @fax4ever !
AtomicBoolean needs for return result from lambda expression
Method "computeIfAbsent" of ConcurentMap needs for return (create or just return) FactoryHolder for required path

Synchronization by ConcurentMap ...It can be excessively ...need think

Reply:
Let me understand the scope of the issue. This issue is made to avoid re-creating a service if I need a second session factory? Let's talk more about the real use case. Please